crates/ra_hir_def/src/item_tree.rs

```diff
@@ -0,0 +1,486 @@
+    structs: ArenaMap<Idx<Struct>, AstPtr<ast::StructDef>>,
+    fields: ArenaMap<Idx<Field>, Either<AstPtr<ast::RecordFieldDef>, AstPtr<ast::TupleFieldDef>>>,
+    unions: ArenaMap<Idx<Union>, AstPtr<ast::UnionDef>>,
+    enums: ArenaMap<Idx<Enum>, AstPtr<ast::EnumDef>>,
+    variants: ArenaMap<Idx<Variant>, AstPtr<ast::EnumVariant>>,
+    consts: ArenaMap<Idx<Const>, AstPtr<ast::ConstDef>>,
+    statics: ArenaMap<Idx<Static>, AstPtr<ast::StaticDef>>,
+    traits: ArenaMap<Idx<Trait>, AstPtr<ast::TraitDef>>,
+    impls: ArenaMap<Idx<Impl>, AstPtr<ast::ImplDef>>,
+    type_aliases: ArenaMap<Idx<TypeAlias>, AstPtr<ast::TypeAliasDef>>,
+    mods: ArenaMap<Idx<Mod>, AstPtr<ast::Module>>,
+    macro_calls: ArenaMap<Idx<MacroCall>, AstPtr<ast::MacroCall>>,
+    exprs: ArenaMap<Idx<Expr>, AstPtr<ast::Expr>>,
+}
+
+impl ItemTree {
+    fn new(syntax: &ast::SourceFile) -> ItemTree {
+        ItemTree::default()
+    }
+}
+
+pub struct Import {
+    pub path: ModPath,
+    pub alias: Option<ImportAlias>,
+    pub visibility: RawVisibility,
+    pub is_glob: bool,
+    pub is_prelude: bool,
+    pub is_extern_crate: bool,
+    pub is_macro_use: bool,
+}
+
+pub struct Function {
+    pub name: Name,
+    pub attrs: Attrs,
+    pub visibility: RawVisibility,
+    pub generic_params: GenericParams,
+    pub has_self_param: bool,
+    pub params: Vec<TypeRef>,
+    pub ret_type: TypeRef,
+}
+
+pub struct Struct {
+    pub name: Name,
+    pub attrs: Attrs,
+    pub visibility: RawVisibility,
+    pub generic_params: GenericParams,
+    pub fields: Fields,
+}
+
+pub struct Union {
+    pub name: Name,
+    pub attrs: Attrs,
+    pub visibility: RawVisibility,
+    pub generic_params: GenericParams,
+    pub fields: Fields,
+}
+
+pub struct Enum {
+    pub name: Name,
+    pub attrs: Attrs,
+    pub visibility: RawVisibility,
+    pub generic_params: GenericParams,
+    pub variants: Range<Idx<Variant>>,
+}
+
+pub struct Const {
+    /// const _: () = ();
+    pub name: Option<Name>,
+    pub visibility: RawVisibility,
+    pub type_ref: TypeRef,
+}
+
+pub struct Static {
+    pub name: Name,
+    pub visibility: RawVisibility,
+    pub type_ref: TypeRef,
+}
+
+pub struct Trait {
+    pub name: Name,
+    pub visibility: RawVisibility,
+    pub generic_params: GenericParams,
+    pub auto: bool,
+    pub items: Vec<AssocItem>,
+}
+
+pub struct Impl {
+    pub generic_params: GenericParams,
+    pub target_trait: Option<TypeRef>,
+    pub target_type: TypeRef,
+    pub is_negative: bool,
+    pub items: Vec<AssocItem>,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct TypeAlias {
+    pub name: Name,
+    pub visibility: RawVisibility,
+    pub generic_params: GenericParams,
+    pub type_ref: Option<TypeRef>,
+}
+
+pub struct Mod {
+    pub name: Name,
+    pub visibility: RawVisibility,
+    pub items: Vec<ModItem>,
+}
+
+pub struct MacroCall {
+    pub name: Option<Name>,
+    pub path: ModPath,
+    pub export: bool,
+    pub builtin: bool,
```

Comment:
I'd suggest to unify the booleans naming, e.g. in `Import` we use `is_*`, but here we don't